libraries/botbuilder-dialogs/src/dialogSet.ts

```diff
@@ -115,19 +116,18 @@ import { DialogContext, DialogState } from './dialogContext';
  * server.post('/api/messages', (req, res) => {
  *     adapter.processActivity(req, res, async (context) => {
  *         // Get conversation state and create DialogContext object
- *         const conversation = conversationState.get(context);
- *         const dc = dialogs.createContext(context, conversation);
+ *         const dc = dialogs.createContext(context);
  *
  *         // Check for any interruptions
  *         const isMessage = context.activity.type === ActivityType.Message;
  *         if (isMessage) {
  *             const utterance = context.activity.text.trim().toLowerCase();
  *             if (utterance.startsWith('edit profile')) {
- *                 await dc.endAll().begin('fillProfile');
+ *                 await dc.cancelAll().begin('fillProfile');
```

Comment:
missing `return` … if you don't return, continued execution may fail with errors. Generally, I found that `begin`, `repeat`, `prompt`, and `end` must be `return` or execution may fail in subsequent turns...

Reply:
it shouldn't cause problems if you don't return here as long as you are checking the dc.responded and other status fields.  that said, I added returns here to simplify things.

Reply:
Yes it will...try running this bot without the `return` for the begin method ;)

Also, you are chaining `dc.cancelAll().begin('fillProfile')`...this will give you an error:
`BotFrameworkAdapter.processActivity(): 500 ERROR - TypeError: dc.cancelAll(...).begin is not a function`

These are not promisable objects so you can't chain them any more ;(